toolchain/check/testdata/facet/no_prelude/combine.carbon

```diff
@@ -136,3 +166,50 @@ fn F() {
   // CHECK:STDERR:
   (({} as C) as (C as (A & B))).((B).G)();
 }
+
+// --- fail_todo_generic_combine.carbon
+
+library "[[@TEST_NAME]]";
+
+import Core;
+
+interface A {
+}
+interface B {
+  fn G();
+}
+
+fn G[T:! A & B](t: T) {
+  // CHECK:STDERR: fail_todo_generic_combine.carbon:[[@LINE+4]]:3: error: cannot access member of interface `B` in type `T` that does not implement that interface [MissingImplInMemberAccess]
+  // CHECK:STDERR:   t.(B.G)();
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR:
+  t.(B.G)();
+}
+
+class C {}
+impl C as A {}
+impl C as B {
+  fn G() {}
+}
+
+fn F() {
+  // CHECK:STDERR: fail_todo_generic_combine.carbon:[[@LINE+17]]:3: error: semantics TODO: `impl lookup for a FacetType with more than one interface` [SemanticsTodo]
+  // CHECK:STDERR:   G({} as C);
+  // CHECK:STDERR:   ^~~~~~~~~~
+  // CHECK:STDERR: fail_todo_generic_combine.carbon:[[@LINE-18]]:1: note: while deducing parameters of generic declared here [DeductionGenericHere]
+  // CHECK:STDERR: fn G[T:! A & B](t: T) {
+  // CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR:
+  // CHECK:STDERR: fail_todo_generic_combine.carbon:[[@LINE+10]]:3: error: cannot implicitly convert from `type` to `A & B` [ImplicitAsConversionFailure]
+  // CHECK:STDERR:   G({} as C);
+  // CHECK:STDERR:   ^~~~~~~~~~
+  // CHECK:STDERR: fail_todo_generic_combine.carbon:[[@LINE+7]]:3: note: type `type` does not implement interface `Core.ImplicitAs(A & B)` [MissingImplInMemberAccessNote]
+  // CHECK:STDERR:   G({} as C);
+  // CHECK:STDERR:   ^~~~~~~~~~
+  // CHECK:STDERR: fail_todo_generic_combine.carbon:[[@LINE-28]]:1: note: while deducing parameters of generic declared here [DeductionGenericHere]
+  // CHECK:STDERR: fn G[T:! A & B](t: T) {
+  // CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR:
```

Comment:
Why does this diagnose twice? Shouldn't it only be one?

Maybe this is related to the TODO, but even once the TODO is addressed, can lookup fail in a way that would cause cascading diagnostics here? The `type` note here is also confusing to me.

Reply:
There will only be the one error here once the TODO is gone, if you tried to pass a type `C` which does not implement the thing you need (e.g. some `D` interface) on the receiving side.

I agree the `type` note is confusing, it's because deduce unwraps things and tries to convert the types instead of the values in this case, so it converts a value `C` of type `type` to a value of type `FacetType(A & B)` which is failing. For normal values, printing the type is good, but when the value _is_ a type it's not great. I thought I filed an issue about this, maybe I just noted it in discord once. I will file an issue now.

Reply:
https://github.com/carbon-language/carbon-lang/issues/5027

Reply:
So in the future `GetInterfaceIdFromConstantId` won't be allowed to diagnose, it will just return a better value, and something else will diagnose as needed?

LG on that understanding

Reply:
Yep! It will be returning multiple interfaces instead of making a TODO error